### PR TITLE
Coop robots action client

### PR DESCRIPTION
**Pull request by jlurgo**

### CODE REVIEW

File: navigation/ca_swarm/scripts/box_client.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        self.move_base.wait_for_server(rospy.Duration(60))
+
+        rospy.loginfo("Connected to move base server")
+        rospy.loginfo("Starting navigation test")
+
+        # Initialize a counter to track waypoints
+        i = 0
+
+        # Cycle through the four waypoints
+        while i < 4 and not rospy.is_shutdown():
```
Comment (eborghi10) [inline navigation/ca_swarm/scripts/box_client.py]: Instead of `4`, use `len(waypoints)`

**Review by eborghi10 — CHANGES_REQUESTED** [review]

File: navigation/ca_swarm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+#############
+
+## Add gtest based cpp test target and link libraries
+# catkin_add_gtest(${PROJECT_NAME}-test test/test_ca_swarm.cpp)
+# if(TARGET ${PROJECT_NAME}-test)
+#   target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
+# endif()
+
+## Add folders to be run by python nosetests
+# catkin_add_nosetests(test)
```
Comment (hidmic) [inline navigation/ca_swarm/CMakeLists.txt]: @jlurgo let's remove unnecessary bits here.

File: navigation/ca_swarm/package.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>ca_swarm</name>
+  <version>0.0.0</version>
+  <description>The ca_swarm package</description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="create@todo.todo">create</maintainer>
```
Comment (hidmic) [inline navigation/ca_swarm/package.xml]: @jlurgo you? @eborghi10 ?

File: navigation/ca_swarm/package.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>TODO</license>
```
Comment (hidmic) [inline navigation/ca_swarm/package.xml]: @eborghi10 ?

File: navigation/ca_swarm/scripts/box_client.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        # Append each of the four waypoints to the list.  Each waypoint
+        # is a pose consisting of a position and orientation in the map frame.
+        waypoints.append(Pose(Point(1, 1, 0.0), Quaternion(0,0,0,1)));
+
+        # Initialize the visualization markers for RViz
+        self.init_markers()
+
+        # Set a visualization marker at each waypoint
+        for waypoint in waypoints:
+            p = Point()
```
Comment (hidmic) [inline navigation/ca_swarm/scripts/box_client.py:30]: @jlurgo unnecessary?

File: navigation/ca_swarm/scripts/box_client.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        rospy.init_node('box_client', anonymous=False)
+
+        rospy.on_shutdown(self.shutdown)
+
+        # Create a list to hold the waypoint poses
+        waypoints = list()
+
+        # Append each of the four waypoints to the list.  Each waypoint
+        # is a pose consisting of a position and orientation in the map frame.
+        waypoints.append(Pose(Point(1, 1, 0.0), Quaternion(0,0,0,1)));
```
Comment (hidmic) [inline navigation/ca_swarm/scripts/box_client.py:23]: @jlurgo consider collapsing the last two lines into: 
```
waypoints = [Pose(Point(1, 1, 0.0), Quaternion(0,0,0,1))]
```

File: navigation/ca_swarm/scripts/box_client.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        # Publisher to manually control the robot (e.g. to stop it)
+        self.cmd_vel_pub = rospy.Publisher('cmd_vel', Twist)
+
+        # Subscribe to the move_base action server
+        self.move_base = actionlib.SimpleActionClient("move_base", MoveBaseAction)
+
+        rospy.loginfo("Waiting for move_base action server...")
+
+        # Wait 60 seconds for the action server to become available
+        self.move_base.wait_for_server(rospy.Duration(60))
```
Comment (hidmic) [inline navigation/ca_swarm/scripts/box_client.py:43]: @jlurgo nit: 60 sec is a bit long, consider 10 sec

File: navigation/ca_swarm/scripts/box_client.py (hunk 1 above)
Comment (hidmic) [reply navigation/ca_swarm/scripts/box_client.py]: which `4`?

File: navigation/ca_swarm/scripts/box_client.py [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            # Send the goal pose to the MoveBaseAction server
+            self.move_base.send_goal(goal)
+
+            # Allow 1 minute to get there
+            finished_within_time = self.move_base.wait_for_result(rospy.Duration(60))
+
+            # If we don't get there in time, abort the goal
+            if not finished_within_time:
+                self.move_base.cancel_goal()
+                rospy.loginfo("Timed out achieving goal")
```
Comment (hidmic) [inline navigation/ca_swarm/scripts/box_client.py:83]: @jlurgo nit: `rospy.logwarn`

**Review by hidmic — COMMENTED** [review]
Left some comments

**Review by eborghi10 — APPROVED** [review]

**Pull request merged**